tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/PitMutationsViewPageObject.java

```diff
@@ -157,12 +158,13 @@ private MutationsTree(ImmutableList<StatusTree> statuses) {
             this.statuses = statuses;
         }
 
-        public void select(PitMutation mutation) {
+        public boolean select(PitMutation mutation) {
             String status = mutation.getStatus().toString();
             for (StatusTree statusTree : statuses) {
                 if (status.equals(statusTree.statusName)) {
-                    statusTree.select(mutation);
-                    return;
+                    if (statusTree.select(mutation)) {
```

Comment:
Why do we need this? Maybe you could leave a comment about that.

Reply:
I'm actually a bit confused. I faced test failures because we were selecting the wrong mutation in the view. I figured out that it was because the selection algorithm was returning too early so I added guards which fixed the tests... but actually those "guards" are absolutely useless??

I don't know what went through my mind and the test failures were likely due to tests being run in the UI thread. I'll revert the changes.